crates/rustc_codegen_spirv/build.rs

```diff
@@ -0,0 +1,59 @@
+//! This custom build script merely checks whether we're compiling with the appropriate Rust toolchain
+
+#[allow(unused_imports)]
+use colored::*;
+use std::process::Command;
+
+/// The required toolchain version for this package. Update accordingly.
+#[cfg(not(feature = "skip-toolchain-check"))]
+const REQUIRED_TOOLCHAIN: &str = "nightly-2022-08-29";
```

Comment:
This not being checked against the top-level `rust-toolchain` file seems potentially fragile.

You could look at `env::get("RUSTUP_TOOLCHAIN")` and maybe error if *it is set* and it's a different value.

`include_str!("../../../rust-toolchain")` might also work great, in terms of both:
* not having to hardcode the date here (potentially the hash could also be moved there as a comment)
* being able to *print* the file to the user as something they can verbatim paste into the project to get things working (it's not just the date, but also the necessary components)

That should work at least for git dependencies (since the whole repo gets downloaded somewher in `~/.cargo/`), but to make sure this can work *in general* (crates.io etc.), you would have to use `cargo publish -p rustc_codegen_spirv --dry-run` (or `cargo package -p rustc_codegen_spirv` directly, that seems less likely to risk actually publishing).

Reply:
I had considered this (roughly), I mainly thought this wouldn't work through crates.io. I can certainly test.

Reply:
Indeed, `cargo package` fails on `include_str!("../../rust-toolchain")` :/